### PR TITLE
When pushing new element onto array, do not free existing data

### PR DESCRIPTION
**Pull request by seanyoung**
This data will be stale and may lead to corruption.

Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

File: src/emit/solana.rs [hunk 1]
```diff
@@ -2367,104 +2387,147 @@ impl<'a> TargetRuntime<'a> for SolanaTarget {
                 "offset_ptr",
             );
 
-            let offset = binary
-                .builder
-                .build_load(offset_ptr, "offset")
-                .into_int_value();
+            let new_string_length = binary.vector_len(val);
```
Comment (LucasSte) [inline src/emit/solana.rs:2390]: I think we can use better naming. `val` sounds like the value to be saved on storage.

**Review by LucasSte — CHANGES_REQUESTED** [review]

File: src/emit/solana.rs (hunk 1 above)
Comment (seanyoung) [reply src/emit/solana.rs:2390]: But `val` *is* the value being saved to storage. I am all for better names, but I don't know what else to call this. Let me know if you have a suggestion.

File: src/emit/solana.rs [hunk 2]
```diff
@@ -2367,104 +2387,147 @@ impl<'a> TargetRuntime<'a> for SolanaTarget {
...
-                .builder
-                .build_call(
-                    binary.module.get_function("account_data_len").unwrap(),
+            let offset = if existing {
+                let offset = binary
+                    .builder
+                    .build_load(offset_ptr, "offset")
+                    .into_int_value();
+
+                let existing_string_length = binary
```
[the remark this answers is not in the document]
Comment (seanyoung) [reply src/emit/solana.rs:2399]: This code stores a string in storage, and replaced an existing string. This line fetches the length of the existing string in storage. If the new will fit, we do not need to do an allocation and we can just memcpy.

[the remark this answers is not in the document]
Comment (seanyoung) [reply src/emit/solana.rs:2399]: I've added a comment explaining that

**Review by LucasSte — APPROVED** [review]

**Pull request merged**